velox/functions/lib/Map.h

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/Expr.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/lib/CheckDuplicateKeys.h"
+
+namespace facebook::velox::functions {
+
+// Returns a map created using the given key/value arrays.
+// It's used for
+// 1. Presto's map function, see documentation at https://prestodb.io/docs/current/functions/map.html#map
+// 2. Spark's map_from_array function, see documentation at https://spark.apache.org/docs/latest/api/sql/#map_from_arrays
+
```

Comment:
nit: use `///` for the comments of a public API; remove the empty line between comment and implementation.